# Graph embedding data

The Supplementary Graph Embedding folder provides gene network embeddings constructed using the STRING database for protein-protein interactions (PPI), along with five sets of alternative replaceable network edge files and pre-trained gene network embeddings, as utilized in the paper.

The **Supplementary Graph Embedding** folder provides gene network embeddings constructed using the STRING database for protein-protein interactions (PPI), along with five sets of alternative network edge files and pre-trained gene network embeddings, as introduced in the paper.


**STRING-PPI (utilzed in the paper)**:   
Cite:  
[1] Yue, Xiang, et al. "Graph embedding on biomedical networks: methods, applications and evaluations." Bioinformatics 36.4 (2020): 1241-1251.   
[2] Mering, Christian von, et al. "STRING: a database of predicted functional associations between proteins." Nucleic acids research 31.1 (2003): 258-261.  

Download: https://github.com/xiangyue9607/BioNEV

**node2vec_PPI**:  
Cite:  
[1] Yue, Xiang, et al. "Graph embedding on biomedical networks: methods, applications and evaluations." Bioinformatics 36.4 (2020): 1241-1251.   
[2] Grover, Aditya, and Jure Leskovec. "node2vec: Scalable feature learning for networks." Proceedings of the 22nd ACM SIGKDD international conference on Knowledge discovery and data mining. 2016.  

Download: https://github.com/xiangyue9607/BioNEV

**Mashup_PPI**  
Cite:  
[1] Yue, Xiang, et al. "Graph embedding on biomedical networks: methods, applications and evaluations." Bioinformatics 36.4 (2020): 1241-1251.   
[2] Choi, Youngduck, Chill Yi-I. Chiu, and David Sontag. "Learning low-dimensional representations of medical concepts." AMIA Summits on Translational Science Proceedings 2016 (2016): 41.  

Download: https://github.com/xiangyue9607/BioNEV

**BioPlex 3.0**  
Cite:  
[1] Ratajczak, Florin, et al. "Speos: an ensemble graph representation learning framework to predict core gene candidates for complex diseases." Nature Communications 14.1 (2023): 7206.  
[2] Huttlin, Edward L., et al. "Dual proteome-scale networks reveal cell-specific remodeling of the human interactome." Cell 184.11 (2021): 3022-3040.  

Download: https://bioplex.hms.harvard.edu/interactions.php  


**HuRI**  
Cite:  
[1] Ratajczak, Florin, et al. "Speos: an ensemble graph representation learning framework to predict core gene candidates for complex diseases." Nature Communications 14.1 (2023): 7206.   
[2] Luck, Katja, et al. "A reference map of the human binary protein interactome." Nature 580.7803 (2020): 402-408.  

Download: http://www.interactome-atlas.org/download  



**Durg-Target Network**  
Cite:  
[1] Yıldırım, Muhammed A., et al. "Drug—target network." Nature biotechnology 25.10 (2007): 1119-1126.   

Download: [Appendix Material 41587_2007_BFnbt1338_MOESM5_ESM.xls](https://www.nature.com/articles/nbt1338)
